tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/ArcHybridTransitionSystem.scala

```
@@ -396,17 +396,23 @@ case class ArcHybridRequestedArcInterpretation(
       case ArcHybridLeftArc(_) =>
         (StackRef(0)(state).headOption, BufferRef(0)(state).headOption) match {
           case (Some(stackFirst), Some(bufferFirst)) =>
-            val otherToken = arcTokens - stackFirst
+            val otherToken = (arcTokens - stackFirst).head
             arcTokens.contains(stackFirst) &&
-              !arcTokens.contains(bufferFirst) && state.stillActive(otherToken.head)
+              !arcTokens.contains(bufferFirst) &&
+              (state.stack.contains(otherToken) ||
+                otherToken == 0 ||
+                (state.bufferPosition > 0 && state.bufferPosition <= otherToken))
           case _ => false
         }
       case ArcHybridRightArc(_) =>
         (StackRef(0)(state).headOption, StackRef(1)(state).headOption) match {
           case (Some(stackFirst), Some(stackSecond)) =>
-            val otherToken = arcTokens - stackFirst
+            val otherToken = (arcTokens - stackFirst).head
```

It can happen only if the Banker sends a RequestedArc constraint that asks for an arc between a token and itself. If that were to happen, it would be a bug, so probably best to fail loudly.